src/Chocolatey.PowerShell/Helpers/EnvironmentHelper.cs

```diff
@@ -166,32 +170,38 @@ public static void SetVariable(PSCmdlet cmdlet, string name, EnvironmentVariable
 
                 cmdlet.WriteDebug($"Registry type for {name} is/will be {registryType}");
 
-                if (string.IsNullOrEmpty(value))
-                {
-                    registryKey.DeleteValue(name, throwOnMissingValue: false);
-                }
-                else
+                if (cmdlet.ShouldProcess(name, $"set {scope} environment variable"))
                 {
-                    registryKey.SetValue(name, value, registryType);
+                    if (string.IsNullOrEmpty(value))
+                    {
+                        registryKey.DeleteValue(name, throwOnMissingValue: false);
+                    }
+                    else
+                    {
+                        registryKey.SetValue(name, value, registryType);
+                    }
                 }
             }
 
             try
             {
-                // Trigger environment refresh in explorer.exe:
-                // 1. Notify all windows of environment block change
-                NativeMethods.SendMessageTimeout(
-                    hWnd: (IntPtr)NativeMethods.HWND_BROADCAST,
-                    Msg: NativeMethods.WM_SETTINGCHANGE,
-                    wParam: UIntPtr.Zero,
-                    lParam: "Environment",
-                    fuFlags: 2,
-                    uTimeout: 5000,
-                    out UIntPtr result);
-
-                // 2. Set a user environment variable making the system refresh
-                var setxPath = string.Format(@"{0}\System32\setx.exe", GetVariable(cmdlet, EnvironmentVariables.SystemRoot, EnvironmentVariableTarget.Process));
-                cmdlet.InvokeCommand.InvokeScript($"& \"{setxPath}\" {EnvironmentVariables.ChocolateyLastPathUpdate} \"{DateTime.Now.ToFileTime()}\"");
+                if (cmdlet.ShouldProcess("Environment variables", "Notify system of changes"))

```

Comment:
Could we be consistent with these `ShouldProcess` calls on the capitalization used?

Future copy/pasted review comments are referring back to this one.

Reply:
Sorted!